btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go

```diff
@@ -100,6 +101,7 @@ func NewStakingEventWatcher(
 		metrics:                 metrics,
 		unbondingTracker:        NewTrackedDelegations(),
 		pendingTracker:          NewTrackedDelegations(),
+		inProgressTracker:       NewTrackedDelegations(),
```

Comment:
Some doc comment would great about what exactly is `inProgressTracker` delegation ?

Reply:
Agreed, added a doc string. 

inProgressTracker is basically a cache that says, hey we are trying to activate this (in a go routine) so let's not try again, hence the check in the for loop, and removal in the go routine from it.